### PR TITLE
[JENKINS-72371] rework node monitor configuration

### PR DESCRIPTION
**Pull request by mawinter69**
This PR reworks the way node monitors are configured. It ensures that also monitors which are set to ignored, can be configured. 

Previously, there was a checkbox before each monitor, where the behaviour was not totally clear. It meant that the monitor is ignored but still collecting data and not disabled as one might think. But when the monitor was ignored any configuration was lost and default values were used. Due to this behaviour the [Versions Node Monitor plugin](https://plugins.jenkins.io/versioncolumn/) implemented its own disconnect checkbox.

Before:
image

After:
image

See [JENKINS-72371](https://issues.jenkins.io/browse/JENKINS-72371).

### Testing done
Before:
1. Go to computer/configure
2. Set Value for Free Temp Space to 2 GiB and 3 GiB warning, apply and reload page
3. Uncheck Free Space Space (the config fields are no longer visible) apply and reload page
4. Check the file nodeMonitors.xml in JENKINS_HOME, there are the default values of 1GiB and 2 GiB warning.
5. Check Free Space Space in the UI, the config fields reappear and contain the defaults

After:
1. Go to computer/configure
2. Set Value for Free Temp Space to 2 GiB and 3 GiB warning, apply and reload page
3. Check `Don't Mark agents temporarily offline` for Free Space Space apply and reload
4. Check the file nodeMonitors.xml in JENKINS_HOME, there values of 2GiB and 3GiB warning are still there
5. The UI shows the configured values and not the defaults

### Proposed changelog entries

- JENKINS-72371, rework node monitor configuration

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [x] The Jira issue, if it exists, is well-described.
- [x] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [x] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@mention

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

**Review by Kevin-CB — COMMENTED** [review]
Looks fine security wise.

**Review by NotMyFault — APPROVED** [review]
looks neat

**Review by timja — APPROVED** [review]
(Removed ready for merge as I see it was flagged as needing ATH)

**Comment by timja** [comment]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
